userguide/tutorials/payment_plugin.adoc

```diff
@@ -1,61 +1,119 @@
-= Developing a Payment Plugin
+= Payment Plugin Overview
 
 Payment plugins integrate Kill Bill with a specific gateway (i.e. payment processor), such as Stripe or Braintree. While we already have many open-source payment plugins available on GitHub, this guide will give you pointers in case you need to develop your own.
 
-Before reading this guide, make sure to familiarize yourself with the main payment userguide.
+Before reading this guide, make sure to familiarize yourself with https://docs.killbill.io/latest/userguide_subscription.html#components-payment[payment components] and the https://docs.killbill.io/latest/userguide_payment.html[payment userguide].
+
+= Developing a Payment Plugin
+
+We provide an interface called https://github.com/killbill/killbill-plugin-api/blob/master/payment/src/main/java/org/killbill/billing/payment/plugin/api/PaymentPluginApi.java[PaymentPluginApi]. This is the interface between Kill Bill and payment plugins. In order to create your own payment plugin, you need to create a class that implements this interface and implement the methods in this interface.
+
+We provide a https://github.com/killbill/killbill-plugin-framework-java[plugin framework] which has many classes that can be used out of the box for plugin development. We strongly encourage you to use this library.  
+
+== Getting Started
+
+We provide a simple https://github.com/killbill/killbill-hello-world-java-plugin[Hello World Plugin] that can be used as the starting point to develop your payment plugin. 
+The *Hello World Plugin* already includes the plugin framework. 
+
+In order to get started with your own payment plugin, you need to do the following:
+
+. Set up the *Hello World Plugin* as per our https://docs.killbill.io/latest/plugin_development.html#_java_plugins[plugin development] document.
+
+. Create a class similar to https://github.com/killbill/killbill-hello-world-java-plugin/blob/719779b1ea25c14928e92996b6aa21cc9bf8d4fe/src/main/java/org/killbill/billing/plugin/helloworld/HelloWorldPaymentPluginApi.java[HelloWorldPaymentPluginApi.java]. This is a reference class that is included in the *Hello World Plugin*. It implements the `PaymentPluginApi` interface.
+
+. Alternatively, you can create a class that extends the https://github.com/killbill/killbill-plugin-framework-java/blob/6bc92fa3f7f091304499fb78df901ecb03fdaf47/src/main/java/org/killbill/billing/plugin/api/payment/PluginPaymentPluginApi.java[PluginPaymentPluginApi.java]. This is a class from the *Kill Bill Plugin framework*. It also implements the `PaymentPluginApi` interface.
```

Comment:
There is added complexity by using `PluginPaymentPluginApi`, as it expects a few extra DAO classes:

```
PluginPaymentPluginApi<RESP_R extends UpdatableRecord<RESP_R>, RESP_T extends Table<RESP_R>, PM_R extends UpdatableRecord<PM_R>, PM_T extends Table<PM_R>>
```

See for example https://github.com/killbill/killbill-stripe-plugin/blob/master/src/main/java/org/killbill/billing/plugin/stripe/StripePaymentPluginApi.java#L88

This signature is very useful as in practice, for production code, you would want to store the request/response from the gateway in the database.

For this guide however (e.g. for the GoCardless example), I would not worry about that / I would skip these details, as this might confuse the reader.

Note that you can still use the killbill-plugin-framework-java library, as it provides implementation of some POJOs, e.g. https://github.com/killbill/killbill-plugin-framework-java/tree/master/src/main/java/org/killbill/billing/plugin/api/payment (I'm just suggesting skipping using the `PluginPaymentPluginApi` for that tutorial).